app/models/repo.rb

```diff
@@ -24,6 +24,9 @@ class Repo < ActiveRecord::Base
 
   CLASS_FOR_DOC_LANGUAGE = { "ruby" => DocsDoctor::Parsers::Ruby::Yard }
 
+  default_scope -> { where(removed_from_github: false) }
+  default_scope -> { where(archived: false) }
```

Comment:
I'm strongly opposed to `default_scope` but enumerating why is difficult. It's just one of those things that I've built up scar tissue from over the years. It saves a few minutes now and really hurts later when debugging etc. 

I'm asking on twitter for alternatives. So far I've got:

- Make it a regular scope and call it everywhere. (not so different than `with_some_issues` which is used heavily https://github.com/codetriage/CodeTriage/blob/aa03ffcd56f57c2b12ffb48a752ae27ec3ea93b7/app/models/repo.rb#L176
- Split up logic into different classes. So you might have a AllRepo  class and an ArchivedRepo class etc.  I'm not sure what we would call the class that ignores deleted and archived, maybe `ActiveRepo` class ? 
- Might get more replies on twitter

I think I would lean towards making it a regular scope and applying it as needed.

Reply:
Thanks for the feedback!
I don't have much experience, will take a look in the tweeter responses and search a little bit.
I will come back soon with a alternative.

Reply:
I read a bit and `default_scope` is really broadly hated in the community.
Changed to a 'normal' scope, it's the simplest alternative.